### ISSUE TITLE
VQLAM: codebook never receives gradients (codebook loss uses the straight-through output)

### ISSUE THREAD

**Issue by rakhimovv** [showlab/Olaf-World#8]
Hi there! I'm not sure whether `VQLAM` is used for any reported results, but it seemed worth flagging just in case.

`VectorQuantizer.forward` returns `(z_q, z, x, indices)`, where `z_q = x + (z - x).detach()` is the straight-through output ([blocks.py#L313-L315](https://github.com/showlab/Olaf-World/blob/3729342d42722e8c2d3ac8db2bc5f53c09d30d08/lam/modules/blocks.py#L313-L315)). `VQLatentActionModel.encode` keeps only `z_q` ([vq_lam.py#L101](https://github.com/showlab/Olaf-World/blob/3729342d42722e8c2d3ac8db2bc5f53c09d30d08/lam/modules/vq_lam.py#L101)), and the codebook loss is computed on it ([vq_lam.py#L143](https://github.com/showlab/Olaf-World/blob/3729342d42722e8c2d3ac8db2bc5f53c09d30d08/lam/modules/vq_lam.py#L143)):

```python
codebook_loss = F.mse_loss(z_q, z_e.detach())
```

Because the codebook lookup `z` only appears inside `.detach()`, this term has no gradient path to `vq.codebook.weight`, and the codebook stays at its initialization. `random_restart()` is not called for this quantizer either; its only caller is `ResidualVectorQuantizer` ([blocks.py#L269](https://github.com/showlab/Olaf-World/blob/3729342d42722e8c2d3ac8db2bc5f53c09d30d08/lam/modules/blocks.py#L269)). The reconstruction loss still goes down, which makes this easy to miss.

We ran 20 training steps of each version from the same initialization and data (16-bit mixed precision, the optimizer from `VQLAM.configure_optimizers`):

| | current code | codebook loss on the raw `z` |
|---|---|---|
| steps where the codebook had a gradient | 0 / 20 | 20 / 20 |
| relative change of the codebook weights | 0 | 6.4% |
| reconstruction MSE, first → last step | 0.116 → 0.033 | 0.116 → 0.033 |

One possible fix is to return `z` from `encode` as well and compute the codebook loss on it:

```python
codebook_loss = F.mse_loss(z_code, z_e.detach())  # z_code: the raw codebook vectors
```

Training `vq_lam`, which has no config yet, also turned up a few smaller problems:

- `configs/lam/lam.yaml` sets `beta`, which `VQLAM.__init__` does not accept;
- with the current code, `ddp_find_unused_parameters_false` ([lam.yaml#L44](https://github.com/showlab/Olaf-World/blob/3729342d42722e8c2d3ac8db2bc5f53c09d30d08/configs/lam/lam.yaml#L44)) fails at the first step because the codebook has no gradient; the `on_after_backward` hook in `vq_model.py` reports the same parameter;
- `VQLAM` logs `train/loss`, so `ModelCheckpoint(monitor: train_loss)` ([lam.yaml#L58](https://github.com/showlab/Olaf-World/blob/3729342d42722e8c2d3ac8db2bc5f53c09d30d08/configs/lam/lam.yaml#L58)) fails at the end of the first epoch.

With the loss fix and a `vq_lam.yaml` that drops `beta` and monitors `train/loss`, `vq_lam` trains under `ddp_find_unused_parameters_false`.
